packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { createContext, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+export interface AccessTokenContextValue {
+  /**
+   * The returned accessToken wheter from the props or from the fetchToken function
+   */
+  accessToken?: string
+  /**
+   * If true, the access token is being fetched
+   */
+  isLoading?: boolean
```

Comment:
```suggestion
  /**
   * If true, the AccessTokenProvider is fetching an access token.
   *
   * The AccessTokenProvider could be fetching an initial access token, or it could be re-fetching the access token.
   */
  isLoading: boolean
```

- `accessToken === undefined && !isLoading` → access token not provided
  - log an error
- `accessToken === undefined && isLoading` → initial fetch
  - show a loader
- `accessToken !== undefined && !isLoading` → have an access token
  - use the access token
- `accessToken !== undefined && isLoading` → re-fetching access token
  - use the access token